api/src/certification/enrolment/domain/services/certification-candidates-ods-service.js

```diff
@@ -158,10 +137,27 @@ async function extractCertificationCandidatesFromCandidatesImportSheet({
       userId: null,
     });
 
+    const cleaCertification = complementaryCertificationsInDB.find(
+      (complementaryCertification) => complementaryCertification.key === ComplementaryCertificationKeys.CLEA,
+    );
     try {
-      candidate.validate(isSco);
+      candidate.validate({ isSco, isCompatibilityEnabled, cleaCertificationId: cleaCertification.id });
       _checkForDuplication(candidate);
     } catch (err) {
+      if (err?.code?.includes('subscriptions')) {
+        if (isCompatibilityEnabled)
+          throw new CertificationCandidatesError({
+            code: CERTIFICATION_CANDIDATES_ERRORS.CANDIDATE_WRONG_SUBSCRIPTIONS_COMPATIBILITY.code,
+            message: 'A candidate cannot have more than one subscription to a certification',
+            meta: { line },
+          });
```

Comment:
J'ai eu du mal à comprendre la mécanique, mais je crois comprendre qu'en cas d'erreur dans le `validate` sur les `subscriptions`, Joi nous renvoit une erreur générique `"subscriptions" does not match any of the allowed types` (ce que nous disent les tests du validate). 

Nous avons plusieurs cas où cette erreur peut être renvoyer :
- **aucune souscriptions**
- lorsque l'on a deux certifications complémentaires
- lorsque l'on a deux certifications complémentaires, l'une d'elles est cléA
- lorsque l'on a deux certifications coeur
- **lorsque l'on a un coeur, et une complémentaire non cléA.**

Et je comprend que dans tous ces cas nous renvoyons un seul et unique message d'erreur  `A candidate cannot have more than one subscription to a certification` ce qui ne répond pas à certains des cas non valides

Est-il possible d'avoir des messages plus précis, directement créer dans le `validate` et gérés par Joi ?

WIP

Reply:
Quitte à faire une autre PR pour ça, je rejoins la remarque de Matthieu, il serait mieux d'avoir une complète maîtrise des messages que l'on retourne lors de cette validation.